apps/aecore/lib/aecore/chain/block_validation.ex

```diff
@@ -8,7 +8,7 @@ defmodule Aecore.Chain.BlockValidation do
   alias Aecore.Chain.ChainState
   alias Aecore.Chain.Difficulty
 
-  @spec validate_block!(Block.block(), Block.block(), map(), list()) :: {:error, term()} | :ok
+  @spec validate_block!(%Block{}, %Block{}, map(), list(%Block{})) :: {:error, term()} | :ok
```

Comment:
Shouldn't we use Block.t()? (add `@type t :: block()` to `Aecore.Structures.Block`  and use it)

Reply:
`Aecore.Structures.Block` does have @type t :: %Block{} [currently](https://github.com/aeternity/elixir-research/pull/182/commits/426bab6bd5887316433c961ecba43f9c985dbc86#diff-ba597a8315c1932793bd18d5cb93f2aeR9) .

A previous comment [here](https://github.com/aeternity/elixir-research/pull/180#discussion_r159038716) suggested that `%Block{}` would be preferable to `Block.t()` . (I like `Block.t` but without the erlang-style parens). I have changed all the  `Aecore.Structures` to using the `%Block{}` style. The team does prefer erlang-style typings, so the endings with `Block.t()` will have parens. The correct typespec style is entirely a team preference. So I'll ask the question again to be sure:

:question: Which style does the team prefer; `%Block{}` or `Block.t()`?

Reply:
lets go with `Block.t()`